src/app/plan/components/Wizard/HooksStep.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useEffect, useState, useContext } from 'react';
+import {
+  Table,
+  TableBody,
+  TableHeader,
+} from '@patternfly/react-table';
+import { Spinner } from '@patternfly/react-core/dist/esm/experimental';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+
+import {
+  Grid,
+  GridItem,
+  Text,
+  TextContent,
+  TextVariants,
+  Bullseye,
+  EmptyState,
+  EmptyStateVariant,
+  EmptyStateBody,
+  Title,
+  Button,
+  EmptyStateIcon,
+  InputGroup,
+  TextInput,
+  Flex,
+  FlexItem,
+  FlexModifiers
+} from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
+import {
+  AddEditMode,
+} from '../../../common/add_edit_state';
+import HooksFormContainer from './HooksFormContainer';
+
+const classNames = require('classnames');
+const styles = require('./HooksStep.module')
+
+const HooksStep = (props) => {
+  const {
+    updateHookRequest,
+    addHookRequest,
+    isFetchingHookList,
+    migHookList,
+    fetchHooksRequest,
+    hookAddEditStatus,
+    currentPlan,
+    removeHookRequest,
+    watchHookAddEditStatus,
+    isAddHooksOpen,
+    setIsAddHooksOpen
+  } = props;
+
+  const [initialHookValues, setInitialHookValues] = useState({});
+
+  useEffect(() => {
+    if (currentPlan) {
+      fetchHooksRequest(currentPlan.spec.hooks);
+
+    }
+  }, []);
+
+  const onAddEditHookSubmit = (hookValues) => {
+    switch (hookAddEditStatus.mode) {
+      case AddEditMode.Edit: {
+        updateHookRequest(hookValues);
+        setIsAddHooksOpen(false)
+        break;
+      }
+      case AddEditMode.Add: {
+        addHookRequest(hookValues);
+        setIsAddHooksOpen(false)
+        break;
+      }
+      default: {
+        console.warn(
+          `onAddEditSubmit, but unknown mode was found: ${hookAddEditStatus.mode}. Ignoring.`);
+      }
+    }
+  };
+
+  const columns = [
+    { title: 'Name' },
+    { title: 'Definition' },
+    { title: 'Run in' },
+    { title: 'Migration step' },
+  ];
+
+  let rows = []
+  let actions = [];
+  if (migHookList.length > 0) {
+    rows = migHookList.map((migHook, id) => {
+      return {
+        cells: [migHook.hookName, migHook.image, migHook.clusterTypeText, migHook.phase],
+      }
+    });
+    actions = [
+      {
+        title: 'Edit',
+        onClick: (event, rowId, rowData, extra) => {
+          const currentHook = migHookList.find((hook) => hook.hookName === rowData.name.title)
+          setInitialHookValues(currentHook)
+          setIsAddHooksOpen(true)
+          watchHookAddEditStatus(rowData.name.title);
+
+        }
+      },
+      {
+        title: 'Delete',
+        onClick: (event, rowId, rowData, extra) => {
+          removeHookRequest(rowData.name.title, rowData["migration-step"].title)
+        }
+      }
+    ]
+
+  }
+  if (rows.length === 0) {
```

Comment:
This is fine, but it might be cleaner to just render the empty state instead of the table altogether when there are no rows, instead of creating a placeholder row. I do that in the other tables (https://github.com/konveyor/mig-ui/blob/master/src/app/plan/components/Wizard/VolumesTable.tsx#L188) and I created a reusable `TableEmptyState` component for it.